### PR TITLE
Cumulus-721 Adding SNS checks in integration test

### PR DESCRIPTION
**Pull request by kkelly51**
**Summary:** Adding SNS check in integration tests

Addresses [CUMULUS-721: Write an integration test suite demonstrating a message being sent to a SNS topic on completed workflow](https://bugs.earthdata.nasa.gov/browse/CUMULUS-721)

## Changes

* Added new lambda and subscription to catch SNS messages sent during IngestGranule integration test. Updated in example/app/config. Redeployment required
* Added test to check the newly store SNS messages

### CODE REVIEW

File: example/lambdas/snsS3Test/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const { S3 } = require('aws-sdk');
+
+async function handler(event) {
```
Comment (laurenfrederick) [inline example/lambdas/snsS3Test/index.js]: Make sure to document this function

File: example/spec/ingestGranule/IngestGranuleSuccessSpec.js [hunk 2]
```diff
@@ -153,13 +157,26 @@ describe('The S3 Ingest Granules workflow', () => {
     });
   });
 
-  describe('the sf-sns-report task has published a sns message and', () => {
-    it('the granule record is added to DynamoDB', async () => {
-      const record = await granuleModel.get({ granuleId: inputPayload.granules[0].granuleId }); 
+  describe('the sf-sns-report task has published a sns message', () => {
```
Comment (laurenfrederick) [inline example/spec/ingestGranule/IngestGranuleSuccessSpec.js]: I think this should be something like 'An SNS success message'

File: example/spec/ingestGranule/IngestGranuleSuccessSpec.js [hunk 3]
```diff
@@ -152,13 +156,26 @@ describe('The S3 Ingest Granules workflow', () => {
...
+    let lambdaOutput;
+    let existCheck;
+
+    beforeAll(async () => {
+      lambdaOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'PostToCmr');
+      executionName = lambdaOutput.cumulus_meta.execution_name;
+      existCheck = await s3ObjectExists({ Bucket: config.bucket, Key: `${config.stackName}/test-output/${executionName}.output` });
+    });
+
+    it('is published on workflow completion', async () => {
```
Comment (bilts) [inline example/spec/ingestGranule/IngestGranuleSuccessSpec.js]: I think it got lost in the shuffle of creating all these sub-tasks, but I used wording for acceptance criteria that could be translated more or less directly into integration test verbiage.  In this case:

"When Cumulus is configured to send a notification on successful workflow completion and an execution succeeds, it sends a notification to the configured SNS topic"
"When Cumulus is configured to send a notification on failed workflow executions and an execution fails, it sends a notification to the configured SNS topic"

File: example/lambdas/snsS3Test/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const { S3 } = require('aws-sdk');
+
+/**
+ * Receives event trigger from SNS and forwards event message to S3 bucket
+ *
+ * @param {*} event - from SNS
```
Comment (scisco) [inline example/lambdas/snsS3Test/index.js]: Could you specify the type as `Object`

File: packages/deployment/app/cloudformation.template.yml [hunk 5]
```diff
@@ -155,7 +155,7 @@ Resources:
       {{else}}
       Endpoint: {{this.endpoint}}
       {{/if}}
-      Protocol: {{@key}}
+      Protocol: lambda
```
Comment (scisco) [inline packages/deployment/app/cloudformation.template.yml]: to make things more flexible in the future, could you add a key:value called `protocol: lambda` to the config and reference it here as `{{this.protocol}}`?

File: packages/deployment/app/cloudformation.template.yml [hunk 6]
```diff
@@ -174,12 +174,12 @@ Resources:
         {{else}}
         - Endpoint: {{this.endpoint}}
         {{/if}}
-          Protocol: {{@key}}
+          Protocol: lambda
```
Comment (scisco) [inline packages/deployment/app/cloudformation.template.yml]: see my above comment

File: packages/deployment/app/cloudformation.template.yml (hunk 5 above)
Comment (kkelly51) [reply packages/deployment/app/cloudformation.template.yml]: Yeah good idea

**Pull request merged**